common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/store/util/LwM2MClientSerDes.java

```diff
@@ -188,7 +180,7 @@ private static Object parseValue(ResourceModel.Type type, JsonElement value) {
             case STRING:
                 return value.getAsString();
             case TIME:
-                return Instant.ofEpochMilli(value.getAsLong());
+                return new Date(value.getAsLong());
```

Comment:
`public void sendWriteCompositeRequest(LwM2mClient client, RpcWriteCompositeRequest rpcWriteCompositeRequest, DownlinkRequestCallback<WriteCompositeRequest, WriteCompositeResponse> callback) {`

Maybe it's also worth checking entry.getValue() for Instan and converting to Date?